cli: Example_demo_locality failed

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9220864?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9220864?buildTab=artifacts#/) on release-23.1 @ [310a949162d4d530677147626bc7a72289c987b0](https://github.com/cockroachdb/cockroach/commits/310a949162d4d530677147626bc7a72289c987b0):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 437362 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-sessions @cockroachdb/server @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25874

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9374857?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9374857?buildTab=artifacts#/) on release-23.1 @ [5d5abe704b0551556c35d51aa673d3162f493aa8](https://github.com/cockroachdb/cockroach/commits/5d5abe704b0551556c35d51aa673d3162f493aa8):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (76.95s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 0 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9381370?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9381370?buildTab=artifacts#/) on release-23.1 @ [5d5abe704b0551556c35d51aa673d3162f493aa8](https://github.com/cockroachdb/cockroach/commits/5d5abe704b0551556c35d51aa673d3162f493aa8):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 469620 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9384307?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9384307?buildTab=artifacts#/) on release-23.1 @ [63b683eab3e0ebfff812bb54fdac936c101d9383](https://github.com/cockroachdb/cockroach/commits/63b683eab3e0ebfff812bb54fdac936c101d9383):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (101.74s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 0 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9394793?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9394793?buildTab=artifacts#/) on release-23.1 @ [63b683eab3e0ebfff812bb54fdac936c101d9383](https://github.com/cockroachdb/cockroach/commits/63b683eab3e0ebfff812bb54fdac936c101d9383):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (110.95s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 0 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9417330?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9417330?buildTab=artifacts#/) on release-23.1 @ [35b724947ac464e6af45ce5193e34035fcacb1ef](https://github.com/cockroachdb/cockroach/commits/35b724947ac464e6af45ce5193e34035fcacb1ef):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 454645 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9432431?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9432431?buildTab=artifacts#/) on release-23.1 @ [2674469608f94b72d8cb6c32ff433dd2ab970252](https://github.com/cockroachdb/cockroach/commits/2674469608f94b72d8cb6c32ff433dd2ab970252):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 460513 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/9469769?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/9469769?buildTab=artifacts#/) on release-23.1 @ [b432e8c20339de5cfa7c811a9ee6f5dc98d15a1e](https://github.com/cockroachdb/cockroach/commits/b432e8c20339de5cfa7c811a9ee6f5dc98d15a1e):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 446340 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9483662?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9483662?buildTab=artifacts#/) on release-23.1 @ [571a1744c72415894dc0e742bd89e74938ea71f4](https://github.com/cockroachdb/cockroach/commits/571a1744c72415894dc0e742bd89e74938ea71f4):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 1 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (79.72s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 1 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9497764?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9497764?buildTab=artifacts#/) on release-23.1 @ [de239a7438f44d382c9aefceb65d9c39911dabd2](https://github.com/cockroachdb/cockroach/commits/de239a7438f44d382c9aefceb65d9c39911dabd2):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 465567 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9508560?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9508560?buildTab=artifacts#/) on release-23.1 @ [de239a7438f44d382c9aefceb65d9c39911dabd2](https://github.com/cockroachdb/cockroach/commits/de239a7438f44d382c9aefceb65d9c39911dabd2):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (87.64s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 0 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9517412?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9517412?buildTab=artifacts#/) on release-23.1 @ [2f96695f75b07c872ec5f146acc1fa198135768f](https://github.com/cockroachdb/cockroach/commits/2f96695f75b07c872ec5f146acc1fa198135768f):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (87.45s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 0 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9537110?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9537110?buildTab=artifacts#/) on release-23.1 @ [a9eb6d220fc0957d4f6f8261d447991e9d03b4ad](https://github.com/cockroachdb/cockroach/commits/a9eb6d220fc0957d4f6f8261d447991e9d03b4ad):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 448686 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9551101?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9551101?buildTab=artifacts#/) on release-23.1 @ [5623212db904b3c2721587b35ab8d17441691996](https://github.com/cockroachdb/cockroach/commits/5623212db904b3c2721587b35ab8d17441691996):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 671240 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9556058?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9556058?buildTab=artifacts#/) on release-23.1 @ [c8a4703f9853a442ac676b44d074b43eb387f60c](https://github.com/cockroachdb/cockroach/commits/c8a4703f9853a442ac676b44d074b43eb387f60c):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (86.13s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 0 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101284 cli: Example_demo_locality failed [C-test-failure O-robot branch-release-23.1.0]
- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9570754?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9570754?buildTab=artifacts#/) on release-23.1 @ [4942d6fec9ec3f502714dab8ea28ac8c9c407dac](https://github.com/cockroachdb/cockroach/commits/4942d6fec9ec3f502714dab8ea28ac8c9c407dac):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 448270 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   Example_demo_locality
I230412 18:10:18.874618 1 cli/testutils.go:267  [-] 216  stopping server and cleaning up CLI test
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101284 cli: Example_demo_locality failed [C-test-failure O-robot branch-release-23.1.0]
- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.Example_demo_locality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9580077?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9580077?buildTab=artifacts#/) on release-23.1 @ [cf84c1c11f2f9c1a4a979033fa0af3befb2ccbcf](https://github.com/cockroachdb/cockroach/commits/cf84c1c11f2f9c1a4a979033fa0af3befb2ccbcf):


```
=== RUN   Example_demo_locality
[demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
*
* ERROR: initialization timeout while waiting for server 0 readiness
*
[demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
--- FAIL: Example_demo_locality (90.07s)
got:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
ERROR: initialization timeout while waiting for server 0 readiness
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
want:
demo --nodes 3 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
demo --nodes 9 --multitenant=false -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1,az=b
2	region=us-east1,az=c
3	region=us-east1,az=d
4	region=us-west1,az=a
5	region=us-west1,az=b
6	region=us-west1,az=c
7	region=europe-west1,az=b
8	region=europe-west1,az=c
9	region=europe-west1,az=d
demo --nodes 3 --multitenant=false --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
node_id	locality
1	region=us-east1
2	region=us-east2
3	region=us-east3
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101284 cli: Example_demo_locality failed [C-test-failure O-robot branch-release-23.1.0]
- #99903 cli: Example_demo_locality failed [C-test-failure O-robot branch-master skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*Example_demo_locality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/BUILD.bazel
+++ b/pkg/cli/BUILD.bazel
@@ -405,6 +405,7 @@ go_test(
         "//pkg/util/log/logpb",
         "//pkg/util/protoutil",
         "//pkg/util/stop",
+        "//pkg/util/syncutil",
         "//pkg/util/timeutil",
         "//pkg/util/tracing",
         "//pkg/util/uuid",

--- a/pkg/cli/demo_locality_test.go
+++ b/pkg/cli/demo_locality_test.go
@@ -17,13 +17,18 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/cli/democluster"
 	"github.com/cockroachdb/cockroach/pkg/security/securityassets"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 )
 
-var _ = skipExample_demo_locality
-
-func skipExample_demo_locality() {
+func Example_demo_locality() {
 	c := NewCLITest(TestCLIParams{NoServer: true})
 	defer c.Cleanup()
+
+	// This is slow under deadlock as it starts a 9-node cluster which
+	// has a very high simulated latency between each node.
+	if syncutil.DeadlockEnabled {
+		return
+	}
 
 	defer democluster.TestingForceRandomizeDemoPorts()()
 

--- a/pkg/cli/democluster/demo_cluster.go
+++ b/pkg/cli/democluster/demo_cluster.go
@@ -100,7 +100,7 @@ type transientCluster struct {
 
 // maxNodeInitTime is the maximum amount of time to wait for nodes to
 // be connected.
-const maxNodeInitTime = 30 * time.Second
+const maxNodeInitTime = 60 * time.Second
 
 // secondaryTenantID is the ID of the secondary tenant to use when
 // --multitenant=true.

--- a/pkg/cli/democluster/demo_cluster_test.go
+++ b/pkg/cli/democluster/demo_cluster_test.go
@@ -145,11 +145,10 @@ func TestTransientClusterSimulateLatencies(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	// This is slow under race as it starts a 9-node cluster which
+	// This is slow under race and deadlock as it starts a 9-node cluster which
 	// has a very high simulated latency between each node.
 	skip.UnderRace(t)
-
-	skip.WithIssue(t, 99768, "flaky test")
+	skip.UnderDeadlock(t)
 
 	demoCtx := newDemoCtx()
 	// Set up an empty 9-node cluster with simulated latencies.

